pyseir/cli.py

```diff
@@ -74,21 +62,97 @@ def _states_region_list(state: Optional[str], default: List[str]) -> List[pipeli
         return [pipeline.Region.from_state(s) for s in default]
 
 
-def _state_only_pipeline(
-    region: pipeline.Region, run_mode=DEFAULT_RUN_MODE, output_interval_days=1, output_dir=None,
-) -> model_fitter.ModelFitter:
-    assert region.is_state()
+@dataclass
+class StatePipeline:
+    """Runs the pipeline for one state and stores the output."""
+
+    region: pipeline.Region
+    infer_df: pd.DataFrame
```

Comment:
I know this is the existing name, but maybe something like `inference_result_df` might be more clear

Reply:
Maybe drop the `inference` reference and use `infection_rate_result_df` instead. Make it more clear that it's supporting that metric?

Reply:
Added a comment. I'm trying to create a minimal PR that does this pipeline refactor... and it isn't very small.